src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -593,6 +584,58 @@ SHAMapStoreImp::clearPrior (LedgerIndex lastRotated)
     if (health())
         return;
 
+    {
+        auto const deleteBatch = setup_.deleteBatch;
+        auto const continueLimit = (deleteBatch + 1) / 2;
+
+        // This does make the sql harder to read
+        std::string const deleteQuery (
+            R"sql(
+            DELETE FROM Validations WHERE LedgerHash IN
```

Comment:
Has this been tested on a system with 1 lot of rows, such as a system in s1 with 1 month of history?